lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -373,6 +373,101 @@
+        type: str
+        default: STANDARD_SSL
+        choices: [ 'STANDARD_SSL', 'ADVANTAGE_SSL', 'UC_SSL', 'EV_SSL', 'WILDCARD_SSL', 'PRIVATE_SSL', 'PD_SSL', 'CODE_SIGNING', 'EV_CODE_SIGNING',
+        'CDS_INDIVIDUAL', 'CDS_GROUP', 'CDS_ENT_LITE', 'CDS_ENT_PRO', 'SMIME_ENT' ]
+        version_added: "2.9"
+
+    entrust_requester_email:
+        description:
+            - The email of the requester of the certificate (for tracking purposes).
+            - This is only used by the C(entrust) provider.
+        type: str
+        version_added: "2.9"
+
+    entrust_requester_name:
+        description:
+            - The name of the requester of the certificate (for tracking purposes).
+            - This is only used by the C(entrust) provider.
+        type: str
+        version_added: "2.9"
+
+    entrust_requester_phone:
+        description:
+            - The phone number of the requester of the certificate (for tracking purposes).
+            - This is only used by the C(entrust) provider.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_user:
+        description:
+            - The username for authentication to the Entrust ECS API.
+            - This is only used by the C(entrust) provider.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_key:
+        description:
+            - The key (password) for authentication to the Entrust ECS API.
+            - This is only used by the C(entrust) provider.
+        type: str
+        version_added: "2.9"
+
+    entrust_api_client_cert_path:
+        description:
+            - The path of the client certificate used to authenticate to the Entrust ECS API.
+            - This is only used by the C(entrust) provider.
+        type: path
+        version_added: "2.9"
+
+    entrust_api_client_cert_key_path:
+        description:
+            - The path of the key for the client certificate used to authenticate to the Entrust ECS API
+            - This is only used by the C(entrust) provider.
+        type: path
+        version_added: "2.9"
+
+    entrust_not_after:
+        description:
+            - The point in time at which the certificate stops being valid.
+            - Time can be specified either as relative time or as absolute timestamp.
+            - Time will always be interpreted as UTC.
+            - Note that only the date (day, month, year) is supported for specifying expiry date of the issued certificate.
+            - The full date-time is adjusted to EST (GMT -5:00) before issuance, which may result in a certificate with an expiration date one day
+              earlier than expected if a relative time is used.
+            - The minimum certificate lifetime is 90 days, and maximum is three years.
+            - Valid format is C([+-]timespec | ASN.1 TIME) where timespec can be an integer
+              + C([w | d | h | m | s]) (e.g. C(+32w1d2h).
+            - Note that if using relative time this module is NOT idempotent.
```

Comment:
The various other providers had the same disclaimer, and similarly don't check before regenerating - I added it because I (perhaps mistakenly) assumed the "not idempotent if relative is used" disclaimer was because the end output is based on the day and time the module runs.

In particular, running with a relative time, you could run 10 minutes apart and get a full day difference in issued cert expiration, since we round to day.

Reply:
Good point; I think the missing check for the self-signed and own-ca providers is a bug. They should check it, and regenerate the certificate if the timestamps are different.

I'm not sure whether you want to do something similar for the `entrust` provider. The ACME provider, for example, does not have something like this, and I don't think it makes much sense when a "real" CA is involved. There, simply regenerating a certificate usually involves some costs, and you're usually more interested in whether your certificate is valid for a long enough timespan from now on.

Reply:
I agree, I think I should skip making judgement calls on existing, valid certificates - it has a different cost and impact for every 'real' CA, so could lead to really inconsistent behavior. If anything, I think standalone do-one-thing ECS specific modules for operations that extend past "I want a fresh cert put <here>" probably makes more sense, because we can document the behavior that is specific to our API, since once you get past "new cert" CAs tend to diverge significantly.

I'll remove the idempotent reference,